odoo/addons/test_read_group/tests/test_date_range.py

```diff
@@ -126,3 +127,141 @@ def test_duplicate_month(self):
+                '__domain': [('date.week', '=', 5.0)]
+            }, {
+                'date:relative_week': 6.0,
+                'date_count': 1,
+                'value': 1,
+                '__domain': [('date.week', '=', 6.0)]
+            }, {
+                'date:relative_week': 22.0,
+                'date_count': 1,
+                'value': 20,
+                '__domain': [('date.week', '=', 22.0)]
+            }])
+        res = self.Model.search(result[0]['__domain'])
+        self.assertEqual(len(res), 1)
+        self.assertEqual(res.mapped('value'), [300])
+
+    def test_full_usecase_quarter(self):
+        result = self.Model.read_group([],
+                                       fields=['date', 'value'],
+                                       groupby=['date:relative_quarter'])
+        self.assertEqual(result, [
+            {
+                'date:relative_quarter': 1.0,
+                'date_count': 2,
+                'value': 301,
+                '__domain': [('date.quarter', '=', 1.0)]
+            }, {
+                'date:relative_quarter': 2.0,
+                'date_count': 1,
+                'value': 20,
+                '__domain': [('date.quarter', '=', 2.0)]
+            }])
+        res = self.Model.search(result[0]['__domain'])
+        self.assertEqual(len(res), 2)
+        self.assertEqual(res.mapped('value'), [1, 300])
+
+    def test_unknown_granularity(self):
+        with self.assertRaises(ValueError):
+            self.Model.read_group([], fields=['date', 'value'], groupby=['date:relative_year'])
+
+        with self.assertRaises(ValueError):
+            self.Model.read_group([], fields=['date', 'value'], groupby=['date:relative_unknown'])
+
+    def test_many2one(self):
+        order = self.env["test_read_group.order"].create({"date": "2021-02-09"})
+        self.env["test_read_group.order.line"].create({"order_id": order.id, "value": 1})
+        result = self.env["test_read_group.order.line"].search([('order_id.date.month', '=', 2)])
+
+        self.assertEqual(result, order.line_ids)
+
+    def test_many2many(self):
+        user = self.env["test_read_group.user"].create({"name": "John"})
+        self.env["test_read_group.task"].create([{"user_ids": [(4, user.id)], "name": "task 1", "date": "2021-02-09"},
+                                                         {"user_ids": [(4, user.id)], "name": "task 2", "date": "2021-04-15"}])
+        result = self.env["test_read_group.user"].search([('task_ids.date.month', '=', 2)])
+        self.assertEqual(result, user)
+
+    def test_related_field(self):
+        order = self.env["test_read_group.order"].create({"date": "2021-02-09"})
+        order_line = self.env["test_read_group.order.line"].create({"order_id": order.id, "value": 1})
+        result = self.env["test_read_group.order.line"].search([('date.month', '=', 2)])
+        self.assertEqual(result, order_line)
+
+    def test_inherit(self):
+        pass
```

Comment:
IMO, it should be in the test_search.py of test_new_api ? 
One test with an `assertQueries` to show the SQL part generated by this  would be nice.